File: backend/src/test/java/app/TemperatureListenerTest.java
package app;

import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.Mock;
import org.mockito.runners.MockitoJUnitRunner;

import static org.mockito.Mockito.verify;

@RunWith(MockitoJUnitRunner.class)
public class TemperatureListenerTest {

    private TemperatureListener temperatureListener;
    @Mock private TemperatureService mockTemperatureService;

    @Before
    public void setUp() throws Exception {
        temperatureListener = new TemperatureListener(mockTemperatureService);
    }

    @Test
    public void process_calls_the_temperatureService() {
        temperatureListener.process("21.39");

        verify(mockTemperatureService).process("21.39");
    }
}

File: backend/src/test/kotlin/app/TemperatureListenerTest.kt
package app

import com.nhaarman.mockito_kotlin.mock
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.mockito.Mockito.verify

class TemperatureListenerTest {

    private lateinit var temperatureListener: TemperatureListener
    private val mockTemperatureService: TemperatureService = mock()

    @BeforeEach
    fun setUp() {
        temperatureListener = TemperatureListener(mockTemperatureService)
    }

    @Test
    fun `process calls the temperatureService`() {
        temperatureListener.process("21.39")

        verify(mockTemperatureService).process("21.39")
    }
}